File: common/src/main/java/com/ssblur/scriptor/registry/words/Subjects.java
package com.ssblur.scriptor.registry.words;

import com.ssblur.scriptor.word.subject.*;

public class Subjects {
  private static final WordRegistry INSTANCE = WordRegistry.INSTANCE;

  public final Subject SELF = INSTANCE.register("self", new SelfSubject());
  public final Subject TOUCH = INSTANCE.register("touch", new TouchSubject());
  public final Subject HITSCAN = INSTANCE.register("hitscan", new HitscanSubject());
  public final Subject PROJECTILE = INSTANCE.register("projectile", new ProjectileSubject());
  public final Subject STORM = INSTANCE.register("storm", new StormSubject());
  public final Subject RUNE = INSTANCE.register("rune", new RuneSubject());
  public final Subject ENCHANT = INSTANCE.register("enchant", new ImbueSubject());
}

File: common/src/main/kotlin/com/ssblur/scriptor/registry/words/Subjects.kt
package com.ssblur.scriptor.registry.words

import com.ssblur.scriptor.registry.words.WordRegistry.register
import com.ssblur.scriptor.word.subject.*

@Suppress("unused")
object Subjects {
  val SELF = register("self", SelfSubject())
  val TOUCH = register("touch", TouchSubject())
  val HITSCAN = register("hitscan", HitscanSubject())
  val PROJECTILE = register("projectile", ProjectileSubject())
  val STORM = register("storm", StormSubject())
  val RUNE = register("rune", RuneSubject())
  val ENCHANT = register("enchant", ImbueSubject())
}
